app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+# Controller for Groups
+class GroupsController < ApplicationController
+  prepend_before_action :set_group, only: %i(show edit update destroy)
+  before_action :set_draw
+
+  def show; end
+
+  def new
+    @group = Group.new(draw: @draw)
+  end
+
+  def create
+    p = group_params.to_h
+    p[:leader_id] = current_user.id unless current_user.admin?
+    result = GroupCreator.new(p).create!
```

Comment:
oooh, turns out YARD has a way to [denote](http://yardoc.org/types.html#see_results) duck types so I'll update it accordingly